app/models/organization.rb

```diff
@@ -138,7 +138,7 @@ def downcase_slug
   def conditionally_update_articles
     return unless Article::ATTRIBUTES_CACHED_FOR_RELATED_ENTITY.detect { |attr| saved_change_to_attribute?(attr) }
 
-    articles.each(&:save)
+    Organizations::SaveAllArticlesWorker.perform_async(id)
```

Comment:
instead of loading all articles to do in one go. it's better to loop the operation fed into a worker that only update 1 article. this way the worker can self balance out the priority instead of being queued one big job

Reply:
and use the following sidekiq bulk method for more efficiency https://andycroll.com/ruby/enqueue-jobs-quickly-with-sidekiq-bulk/